lib/array.rb

```diff
@@ -0,0 +1,31 @@
+class Array
+  def gusject(*args, &block)
+    if args.empty? || args[0].instance_of?(Symbol)
+      gusject_no_initial(args, block)
+    else
+      gusject_with_initial(args, block)
+    end
+  end
+
+  private
+
+  def gusject_with_initial(args, block)
+    sum = args[0]
+    if block
+      each { |el| sum = block.call(sum, el) }
```

Comment:
Can you quickly explain to me here how you've gotten round using `drop(1).each` without accidentally doubling the initial element in `args`?